src/lib/support/Nullable.h

```diff
@@ -0,0 +1,132 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/support/MatterScalarNullHandling.h>
+#include <optional>
+#include <type_traits>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace DataModel {
```

Comment:
This is the first time we introduce a Datamodel (and app:: as well) into lib/support.

I am unsure how we should approach things here ... I feel uneasy about having app:: and others into lib/support however changing that introduces more blast radius. Could we set up aliases instead in some way?